logilica_weekly_report/update_gdoc.py

```diff
@@ -1,12 +1,46 @@
-def update_gdoc(pdf_items: dict[str, dict[str, any]], config: dict[str, any]) -> None:
+import base64
+from typing import Callable
+
+from yattag import Doc, SimpleDoc
+
+
+def update_gdoc(pdf_items: dict[str, dict[str, bytes]], config: dict[str, any]) -> None:
     """Take the set of charts and text extracted from the PDF and insert it
     into the Google Doc.
+
+    This is accomplished by generating an HTML document and uploading it to
+    Google Drive with a MIME type of `'application/vnd.google-apps.document'`,
+    which causes Google to treat/convert it to a GDoc.  The images are inserted
+    into it as data URIs, which Google will manage appropriately.
     """
 
-    # Debug stand-in code
-    for team, i in pdf_items.items():
-        for dashboard, j in i.items():
-            for name, content in j["images"].items():
-                print(f"{team} {dashboard} {name}:  {content.width}x{content.height}")
-            for k, text in enumerate(j["text"]):
-                print(f"{team} {dashboard} text blob #{k}:  {len(text)} characters")
+    doc, tag, text = Doc().tagtext()
+
+    doc.asis("<!DOCTYPE html>")
+    with tag("html"):
+        with tag("body"):
+            with tag("h1"):
+                text("Logilica Weekly Report")
+                add_teams(pdf_items, doc, tag, text)
+
+    print(doc.getvalue())
```

Comment:
I would expect this to be dumped into a file passed as an argument instead of logging it to stdout. Maybe the function could simply return the string?

Reply:
Actually, the next PR will dump it into a GDoc (once I get all the wrinkles smoothed out).  This PR (as I tried to explain in the description) is just the next step toward that.